service/src/main/java/bio/terra/cbas/controllers/RunSetsApiController.java

```diff
@@ -323,8 +332,20 @@ public ResponseEntity<RunSetStateResponse> postRunSet(RunSetRequest request) {
         new RunSetStateResponse().runSetId(runSetId).runs(runStateResponseList).state(runSetState);
 
     captureResponseMetrics(response);
-    bardService.logRunSetEvent(request, userToken);
-
+    if (bardServerConfiguration.enabled()) {
+      List<String> workflowIds =
+          runStateResponseList.stream()
+              .map(RunStateResponse::getEngineId)
+              .filter(Objects::nonNull)
+              .toList();
+      GithubMethodDetails githubMethodDetails = null;
+      if (methodSourceEnum.equals(PostMethodRequest.MethodSourceEnum.GITHUB)) {
+        githubMethodDetails =
+            githubMethodDetailsDao.getMethodSourceDetails(methodVersion.getMethodId());
+      }
```

Comment:
This mechanism is changing in https://github.com/DataBiosphere/cbas/pull/279 to always just being available already in the `Method`, so no extra "fetch" hops will be required.

Just something to be aware of depending on which order these merge in 😄